OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -147,22 +146,15 @@ public override bool Tick(Actor self)
 
 			var minimumRange = attackAircraft.Info.AttackType == AirAttackType.Strafe ? WDist.Zero : attackAircraft.GetMinimumRangeVersusTarget(target);
 
-			// When strafing we must move forward for a minimum number of ticks after passing the target.
-			if (remainingTicksUntilTurn > 0)
-			{
-				Fly.FlyTick(self, aircraft, aircraft.Facing, aircraft.Info.CruiseAltitude);
-				remainingTicksUntilTurn--;
-			}
-
 			// Move into range of the target.
-			else if (!target.IsInRange(pos, lastVisibleMaximumRange) || target.IsInRange(pos, minimumRange))
+			if (!target.IsInRange(pos, lastVisibleMaximumRange) || target.IsInRange(pos, minimumRange))
 				QueueChild(aircraft.MoveWithinRange(target, minimumRange, lastVisibleMaximumRange, target.CenterPosition, Color.Red));
 
 			// The aircraft must keep moving forward even if it is already in an ideal position.
 			else if (!aircraft.Info.CanHover || attackAircraft.Info.AttackType == AirAttackType.Strafe)
 			{
-				Fly.FlyTick(self, aircraft, aircraft.Facing, aircraft.Info.CruiseAltitude);
-				remainingTicksUntilTurn = attackAircraft.Info.AttackTurnDelay;
+				QueueChild(new Fly(self, target, target.CenterPosition));
+				QueueChild(aircraft.MoveWithinRange(target, lastVisibleMaximumRange, lastVisibleMaximumRange * 2, target.CenterPosition));
```

Comment:
The `lastVisibleMaximumRange * 2` here is a bit arbitrary. Conceptually it should be `WDist.Infinite`, but that doesn't exist. In practice, it doesn't really matter as long as the annulus is bigger than the aircraft can traverse in a single tick.

Reply:
Imho it would be a good idea to place a comment here explaining that (since it does not seem obvious to me, and considering you gave a comment on GitHub you probably think that it's not too obvious as well).

Reply:
I'm still hoping for a bit more elegant solution, but if that cannot be found, I will put the comment in.

Reply:
I see now that we do have `WDist.MaxValue`, which is as close as you can get to infinite with integers. That should make the intention clear.